sw/device/silicon_creator/lib/drivers/retention_sram.h

```diff
@@ -21,10 +24,19 @@ typedef struct retention_sram_creator {
    * ROM.
    */
   uint32_t reset_reasons;
+  /**
+   * Boot services message area.
+   *
+   * This is the shared buffer through which ROM_EXT and silicon owner code
+   * communicate with each other.
+   */
+  boot_svc_msg_t boot_svc_msg;
   /**
    * Space reserved for future allocation by the silicon creator.
    */
-  uint32_t reserved[2048 / sizeof(uint32_t) - 2];
+  uint32_t reserved[(2048 - sizeof(uint32_t) - sizeof(uint32_t) -
```

Comment:
Nit: maybe add a comment to explain the calculation?

Before this change, it looks like we're trying to pad up to 2048 32-bit words. But why did it subtract two?

Needless to say, I also am not following the new version. I'm kind of surprised that the union size is in the numerator, getting divided by sizeof(uint32_t).

Reply:
this is the size of the reserved area at the end. there is an overhead of 1 word because of the version field that comes before creator area. I'll add a comment.

We device by sizeof(uint32_t) because the reserved area in an array of `uint32_t`s.